### PR TITLE
Persist enum symbol info in BIR

### PR DESCRIPTION
**Pull request by pubudu91**
## Purpose
This PR adds enum symbol info to the BIR. So that enums coming from a BIR source can be identified at the semantic API level and create enum symbols for them. 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java [hunk 1]
```diff
@@ -295,6 +297,24 @@ public void visit(BUnionType bUnionType) {
         for (BType memberType : bUnionType.getMemberTypes()) {
             writeTypeCpIndex(memberType);
         }
+
+        if (tsymbol instanceof BEnumSymbol) {
+            buff.writeBoolean(true);
+            writeEnumSymbolInfo((BEnumSymbol) tsymbol);
+        } else {
+            buff.writeBoolean(false);
+        }
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java:306]: @Kishanthan I'm changing the format of the binary a bit here. Do we need to update the kaitai file as well?

Comment (Kishanthan) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java:306]: @pubudu91 yes :)

**Review by Kishanthan — COMMENTED** [review]
@pubudu91 , lets add a test case in BIR spec module too - https://github.com/ballerina-platform/ballerina-lang/tree/master/docs/bir-spec/src/test

**Comment by pubudu91** [comment]
@Kishanthan Added an enum to the type def test file in https://github.com/ballerina-platform/ballerina-lang/pull/28294/commits/352eba9c70bcbf66056ed4652098a0dd6636b30c. Will that be enough?

**Comment by Kishanthan** [comment]
> @Kishanthan Added an enum to the type def test file in [352eba9](https://github.com/ballerina-platform/ballerina-lang/commit/352eba9c70bcbf66056ed4652098a0dd6636b30c). Will that be enough?

You can add a test case to verify this. If you can check the existing test method, you can add an entry to verify the newly added enum here - https://github.com/ballerina-platform/ballerina-lang/blob/master/docs/bir-spec/src/test/java/org/ballerinalang/birspec/BIRTestUtils.java#L89

**Comment by pubudu91** [comment]
> > @Kishanthan Added an enum to the type def test file in [352eba9](https://github.com/ballerina-platform/ballerina-lang/commit/352eba9c70bcbf66056ed4652098a0dd6636b30c). Will that be enough?
> 
> You can add a test case to verify this. If you can check the existing test method, you can add an entry to verify the newly added enum here - https://github.com/ballerina-platform/ballerina-lang/blob/master/docs/bir-spec/src/test/java/org/ballerinalang/birspec/BIRTestUtils.java#L89

Internally, an enum is just another type def. So it should be validated through the `assertTypeDefs()` method. That test did fail when the spec was modified.

**Review by Kishanthan — APPROVED** [review]

**Pull request merged**